subprojects/signing/src/main/java/org/gradle/plugins/signing/Sign.java

```diff
@@ -116,7 +116,7 @@ public Object call() throws Exception {
                 name = "_" + name.substring(1);
             }
 
-            filesWithPotentialNameCollisions.put(name, signature.getToSign());
+            filesWithPotentialNameCollisions.put(name, signature.getSignatureType().fileFor(signature.getToSign()));
```

Comment:
:x: Isn't `signature.getFile()` correct? `signatureType.fileFor(toSign)` seems to be only the default: https://github.com/gradle/gradle/blob/8a7b8ea7bfd2c38689c8fa339c66485f610c10a9/subprojects/signing/src/main/java/org/gradle/plugins/signing/Signature.java#L316-L334

Reply:
Interestingly enough, `getFile()` is not used when generating the signature file:

https://github.com/gradle/gradle/blob/3d3584f20d835168a787bab9e6cd66e5c6956889/subprojects/signing/src/main/java/org/gradle/plugins/signing/type/AbstractSignatureType.java#L38-L45

While I could not find any references to `setFile()`, it looks like `Signature` should pass its `file` to `SignatureType.sign()` instead of ignoring its return value.

Reply:
Interesting, so this never seemed to work as the Javadoc suggests. Instead of deprecating `sign(Signature, File)`, how about we deprecate `setFile`? It has no effect anyway, so we can directly make it a noop, no?
WDYT?

Reply:
Well, since `getFile()` is used for uploading, I think `Signature` should pass its `file` to `SignatureType.sign()` in any case. Thus, we might as well leave `setFile()` in place since it's a public API. Are you okay with that?

Reply:
I think we agree that `getFile` needs to return the signed file, no matter how `Signature` is configured. Like you said, this is used for uploading.

Currently, `setFile` would change the value of `getFile`, but `getFile` would not point anymore to the file generated by `SignatureType.sign`. That is a problem.

So either
1) we extend our public API and add a method to `SignatureType` to fix the behavior when mutating `Signature` or
2) we remove a method (i.e. deprecate for now) from our public API so the destination file of `Signature` cannot be changed any more.

Since no one could have changed the destination of the `Signature` via `setFile` anyway (since it had no effect), I would opt for 2). If the feature to change the destination file for the signature is required at some point, we could re-add it.

WDYT?